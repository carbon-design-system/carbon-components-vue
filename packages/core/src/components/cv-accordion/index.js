import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
