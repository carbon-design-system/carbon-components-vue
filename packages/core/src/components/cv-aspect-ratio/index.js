import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
