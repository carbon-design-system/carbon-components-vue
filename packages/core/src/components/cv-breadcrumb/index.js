import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
