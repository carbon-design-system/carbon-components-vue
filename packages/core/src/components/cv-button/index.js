import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
