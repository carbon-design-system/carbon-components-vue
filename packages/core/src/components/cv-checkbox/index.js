import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
