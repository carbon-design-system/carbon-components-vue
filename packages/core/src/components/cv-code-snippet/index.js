import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
