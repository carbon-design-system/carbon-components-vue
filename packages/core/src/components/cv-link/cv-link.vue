<template>
  <component
    :is="tagType"
    v-on="$listeners"
    v-bind="linkProps"
    class="cv-link"
    :class="[`${carbonPrefix}--link`, { [`${carbonPrefix}--link--inline`]: inline }]"
  >
    <slot></slot>
  </component>
</template>

<script>
import { linkMixin, carbonPrefixMixin } from '../../mixins';

export default {
  name: 'CvLink',
  mixins: [linkMixin, carbonPrefixMixin],
  props: {
    inline: Boolean,
  },
};
</script>
