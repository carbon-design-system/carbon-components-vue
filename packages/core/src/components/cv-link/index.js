import CvLink from './cv-link';

export { CvLink };
