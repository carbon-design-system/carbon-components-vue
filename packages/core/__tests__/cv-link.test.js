// import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow } = awaitNextTick;
import { CvLink } from '@/components/cv-link';

describe('CvLink', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvLink, ['inline', 'disabled'], Boolean);
  testComponent.propsAreType(CvLink, ['href'], String);
  testComponent.propsAreType(CvLink, ['to'], [String, Object]);

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render correctly when it is not `inline`', async () => {
    const propsData = { inline: false };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when it is `inline`', async () => {
    const propsData = { inline: true };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `disabled`', async () => {
    const propsData = { disabled: true };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // The test below gives the following warning:
  // [Vue warn]: Unknown custom element: <router-link> - did you register the component correctly?
  // Could not get rid of it using localVue and vue-router or stubs. That's why it is commented out.
  // it('should render correctly when `router-link` is used', async () => {
  //   const propsData = { to: '/test' };
  //   const wrapper = await shallow(
  //     CvLink,
  //     { propsData },
  //     {
  //       stubs: ['router-link'],
  //     }
  //   );
  //   expect(wrapper.html()).toMatchSnapshot();
  // });

  it('should render correctly when `<a>` is used', async () => {
    const propsData = { href: '/test' };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************

  it('`linkProps` should be computed correctly when `disabled`', async () => {
    const propsData = { href: '/test', to: '/test', disabled: true };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.vm.linkProps).toEqual({ 'aria-disabled': true });
  });

  // The test below gives the following warning:
  // [Vue warn]: Unknown custom element: <router-link> - did you register the component correctly?
  // Could not get rid of it using localVue and vue-router or stubs. That's why it is commented out
  // it('`linkProps` should be computed correctly when `to` is specified and `href` not', async () => {
  //   const to = '/test';
  //   const propsData = { to };
  //   const wrapper = await shallow(CvLink, { propsData });
  //   expect(wrapper.vm.linkProps).toEqual({ to });
  // });

  it('`linkProps` should be computed correctly when `href` and `to` are specified', async () => {
    const to = '/test';
    const href = '/nottest';
    const propsData = { to, href };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.vm.linkProps).toEqual({ href });
  });

  it('`tagType` should be computed correctly when `href` and `to` are specified', async () => {
    const to = '/test';
    const href = '/nottest';
    const propsData = { to, href };
    const wrapper = await shallow(CvLink, { propsData });
    expect(wrapper.vm.tagType).toEqual('a');
  });

  // The test below gives the following warning:
  // [Vue warn]: Unknown custom element: <router-link> - did you register the component correctly?
  // Could not get rid of it using localVue and vue-router or stubs. That's why it is commented out
  // it('`tagType` should be computed correctly when `to` is specified and `href` not', async () => {
  //   const to = '/test';
  //   const propsData = { to };
  //   const wrapper = await shallow(CvLink, { propsData });
  //   expect(wrapper.vm.tagType).toEqual('router-link');
  // });
});
