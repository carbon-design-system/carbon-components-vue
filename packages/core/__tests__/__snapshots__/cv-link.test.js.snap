// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvLink should render correctly when \`<a>\` is used 1`] = `<a href="/test" class="cv-link bx--link"></a>`;

exports[`CvLink should render correctly when \`disabled\` 1`] = `<a aria-disabled="true" class="cv-link bx--link"></a>`;

exports[`CvLink should render correctly when it is \`inline\` 1`] = `<a class="cv-link bx--link bx--link--inline"></a>`;

exports[`CvLink should render correctly when it is not \`inline\` 1`] = `<a class="cv-link bx--link"></a>`;
